.github/workflows/main.yml

```diff
@@ -30,3 +30,36 @@ jobs:
         run: python -m pip install -U tox
       - name: Run lint
         run: tox -elint
+  build:
+    runs-on: ubuntu-latest  # windows-latest | macos-latest
+    name: Test changed-files
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          fetch-depth: 0  # OR "2" -> To retrieve the preceding commit.
```

Comment:
This clones the full history right? That's fine but will add more check out time to each job to clone the history as the repo grows. I feel like we only need a subset to determine the diff against the main branch.

Reply:
True, I can make it `fetch-depth 2` so that it only retrieves the preceding commit.